core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala

```diff
@@ -13,72 +15,76 @@ import akka.persistence.cassandra.journal.TagWriter.TagProgress
 import akka.persistence.cassandra.journal.TagWriters.{ AllFlushed, FlushAllTagWriters, TagWritersSession }
 import akka.persistence.cassandra.journal._
 import akka.persistence.cassandra.query.EventsByPersistenceIdStage.RawEvent
+import akka.persistence.cassandra.query.EventsByPersistenceIdStage.Extractors.Extractor
 import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
 import akka.persistence.cassandra.session.scaladsl.CassandraSession
 import akka.persistence.query.PersistenceQuery
-import akka.serialization.Serialization
+import akka.serialization.SerializationExtension
 import akka.stream.{ ActorMaterializer, OverflowStrategy }
 import akka.stream.scaladsl.{ Sink, Source }
 import akka.util.Timeout
 import akka.{ Done, NotUsed }
 import com.datastax.driver.core.Row
 import com.datastax.driver.core.utils.Bytes
-
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
 object EventsByTagMigration {
   def apply(system: ActorSystem): EventsByTagMigration = new EventsByTagMigration(system)
+
   // Extracts a Cassandra Row, assuming the pre 0.80 schema into a [[RawEvent]]
-  val RawPayloadOldTagSchema = (bucketSize: BucketSize, transportInformation: Option[Serialization.Information]) => (row: Row, ed: EventDeserializer, serialization: Serialization) => {
-    // Get the tags from the old location i.e. tag1, tag2, tag3
-    val tags: Set[String] =
-      if (ed.hasOldTagsColumns(row)) {
-        (1 to 3).foldLeft(Set.empty[String]) {
-          case (acc, i) =>
-            val tag = row.getString(s"tag$i")
-            if (tag != null) acc + tag
-            else acc
+  def rawPayloadOldTagSchemaExtractor(bucketSize: BucketSize, ed: EventDeserializer, system: ActorSystem): Extractor[RawEvent] =
+    new Extractor[RawEvent](ed, SerializationExtension(system)) {
+      override def extract(row: Row, async: Boolean)(implicit ec: ExecutionContext): Future[RawEvent] = {
```

Comment:
changed to a class when adding the `async: Boolean` because I thought the function style Extractor became too difficult to comprehend

The purpose of the `async: Boolean` is that when parallelism=1 then Future.successful otherwise Future().